pages/data/index.vue

```diff
@@ -287,15 +285,67 @@ export default {
     if (!this.$route.query.type) {
       const firstTabType = compose(propOr('', 'type'), head)(searchTypes)
 
-      this.$router.replace({ query: { type: firstTabType } }).then(() => {
-        this.fetchFromContentful()
-      })
+      this.$router.replace({ query: { type: firstTabType } })
     } else {
-      this.fetchFromContentful()
+      this.fetchResults()
     }
   },
 
   methods: {
+    /**
+     * Figure out which source to fetch results from based on the
+     * type of search
+     */
+    fetchResults: function() {
+      const source = propOr('contentful', 'dataSource', this.searchType)
+
+      const searchSources = {
+        contentful: this.fetchFromContentful,
+        blackfynn: this.fetchFromBlackfynn
+      }
+
+      if (typeof searchSources[source] === 'function') {
+        searchSources[source]()
+      }
+
+      // Fetch filters for the search type
+      this.fetchFilters()
+    },
+
+    /**
+     * Get Search results
+     * This is using fetch from the Blackfynn API
+     */
+    fetchFromBlackfynn: function() {
+      this.isLoadingSearch = true
+
+      const searchType = pathOr('', ['query', 'type'], this.$route)
+      let url = `${process.env.discover_api_host}/search/${searchType}s?offset=${this.searchData.skip}&limit=${this.searchData.limit}&organization=SPARC%20Consortium`
+
+      if (searchType === 'file') {
+        url += '&fileType=tiff'
+      }
+
+      const query = pathOr('', ['query', 'q'], this.$route)
+      if (query) {
+        url += `&query=${query}`
+      }
+
```

Comment:
I see, this is why you can't make it a computed property, because of the query bit.

Reply:
Though I still feel like this logic could still be implemented in a computed prop, you'd just return a different url based on whether the `searchType` or `query` condition applies.

Reply:
@mahamzayed I moved this to a computed property, and all tests are passing.